Editor/StatusBox.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace AmazonGameLift.Editor
+{
+    public class StatusBox : VisualElement
+    {
+        public new class UxmlFactory : UxmlFactory<StatusBox> { }
+
+        private readonly VisualElement _container;
+        private Label _statusTextLabel;
+        private bool ShowElement { get; set; }
+        
+        private const string  HiddenClassName = "hidden";
+        private const string  SuccessBoxElementClass = "status-box--success";
+        private const string  InfoBoxElementClass = "status-box--info";
+        private const string  WarningBoxElementClass = "status-box--warning";
+        private const string  ErrorBoxElementClass = "status-box--error";
+
+        private StatusBoxType _currentType;
+
+        private readonly IReadOnlyDictionary<StatusBoxType, string> _statusBoxClasses =
+            new Dictionary<StatusBoxType, string>()
+            {
+                { StatusBoxType.Success, SuccessBoxElementClass },
+                { StatusBoxType.Info, InfoBoxElementClass },
+                { StatusBoxType.Warning, WarningBoxElementClass },
+                { StatusBoxType.Error, ErrorBoxElementClass }
+            };
+
+        private ElementLocalizer _elementLocalizer;
+        private readonly Button _externalButton;
+        private readonly Label _externalButtonLabel;
+
+        public StatusBox()
+        {
+            var asset = Resources.Load<VisualTreeAsset>("EditorWindow/Components/StatusBox");
+            asset.CloneTree(this);
+
+            _container = this.Q("StatusBox");
+            _statusTextLabel = _container.Q<Label>("StatusBoxLabel");
+            _externalButton = this.Q<Button>("StatusBoxExternalButton");
+            _externalButtonLabel = this.Q<Label>("StatusBoxExternalButtonLabel");
+            _elementLocalizer = new ElementLocalizer(_container);
+            this.Q<Button>("StatusBoxCloseButton").RegisterCallback<ClickEvent>(_ =>
+            {
+                Close();
+            });
+
+            UpdateStatusBoxesState();
+        }
+
+        private void AddExternalButton(string externalButtonLink, string externalButtonText)
+        {
+            if (!string.IsNullOrWhiteSpace(externalButtonText))
+            {
+                _externalButtonLabel.text = externalButtonText;
+                _externalButton.RegisterCallback<ClickEvent>(_ => Application.OpenURL(externalButtonLink));
+                _externalButton.RemoveFromClassList(HiddenClassName);
+            }
+        }
+        
+        public void Show(StatusBoxType statusBoxType, string text, string externalButtonLink = "", string externalButtonText = "")
```

Comment:
does c# accept null as a valid string type? Personal preference but i like to use `null`to determine that a particular parameter hasn't been passed in rather than ""

Reply:
Fixed